h/schemas/oauth.py

```diff
@@ -25,23 +22,15 @@ def __init__(self):
 class OAuth2RedirectSchema:
     """Schema for validating OAuth 2 authorization redirect requests."""
 
-    def __init__(self, request: Request, session_key: str) -> None:
-        self._request = request
-        self.state_session_key = f"{session_key}.state"
-
     class OAuth2RedirectData(TypedDict):
         """Return type for OAuth2RedirectSchema.validate()."""
 
         code: str
         state: str
 
-    def validate(self, data: dict[str, Any]) -> OAuth2RedirectData:
-        """Validate `data` and return the validated copy.
-
-        This includes validating the `state` param against a copy that was
-        previously stored in the Pyramid session. Any existing state param in
-        the Pyramid session will be deleted.
-        """
+    @staticmethod
+    def validate(data: dict[str, Any], expected_state) -> OAuth2RedirectData:
```

Comment:
Added the `expected_state` arg here: the view is going to read the state from the session and pass it in to `validate()`